Standards/scs-0300-v1-requirements-for-sso-identity-federation.md

```diff
@@ -0,0 +1,221 @@
+federation. In this configuration it can work as a point of protocol
+transition between different supported SSO protocols (SAML, OAuth 2.0, etc.).
+
+Beyond this capability of using other IdPs as identity sources, it also supports
+using classic LDAP based IAM services as backend (OpenLDAP and Active Directory,
+e.g.).
+
+Keycloak's implementation makes some design decisions, that are specific
+to it and have consequences for clients of the service. E.g. Keycloak
+has a concept of management "Realms", which have their own specific
+set of HTTP API entrypoints, both for administration as well as for IdP
+requests.
+
+Commonly Keycloak realms can be used to map them 1:1 to user domains,
+but since Keycloak supports configuring multiple backend IdPs in a
+realm to be used for "Identity Brokering", there is always the
+possibility to create a kind of "proxy" realm to provide a single
+standard set of HTTP API endpoints for SSO clients (service providers)
+to avoid the need to frequently extend/reduce client service configuration
+whenever a new IdP federation needs to be added to Keycloak to onboard
+a new customer. This is relevant for services like OpenStack Keystone,
+which currently cannot be easily reconfigured for new SSO endpoints
+without restarting the service, making the service unavailable for
+a short span of time and increasing risk connected with service restarts.
+
+Since version 17, Keycloak claims that it's capability for
+"cloud native" deployments on Kubernetes has improved.
+
+[Keycloak is offering a REST API](https://www.keycloak.org/docs-api/20.0.1/rest-api/index.html)
+for all aspects of its administration interface.
+
+For storage of Keycloak configuration and local user metadata
+(e.g. from which external IdP a user account originally came from)
+Keycloak supports several SQL backends through JDBC. Thus
+it can be hooked up to a Postgres Database or to a
+MariaDB/Galera cluster e.g..
+
+### Zitadel
+
+Zitadel is a newer implementation of an SSO IdP.
+It is implemented in Go and under active development and maintained by ZITADEL.
+
+The project is open for community [contributions](https://github.com/zitadel/zitadel/blob/main/CONTRIBUTING.md)
+to all parts of the eco system.
+Feature requests and bugs being tracked on [Github](https://github.com/orgs/zitadel/projects/2/views/5) for development.
+Community questions can be asked in the [public chat](https://zitadel.com/chat) or via [Github Discussions](https://github.com/zitadel/zitadel/discussions).
+ZITADEL offers support for the commonly used authentication and authorization protocols such as OIDC, OAuth2, SAML2.
+It is a compliant and certified OpenID Connect provider with support for various Grant Types for both human users and machine users.
+Compared to Keycloak SPIs, ZITADEL offers Actions to customize and integrate (eg, calling external APIs, Webhooks, customizing pre-built workflows, customizing tokens)
```

Comment:
What does "SPIs" mean? Maybe "Single point of identity"?

Reply:
Ah, nevermind, found it. Seems to be a Keycloak specific term: ["Service Provider Interfaces"](https://www.keycloak.org/docs/latest/server_development/#_providers)